msal/src/test/java/com/microsoft/identity/client/e2e/tests/network/nativeauth/SignUpEmailOTPTest.kt

```diff
@@ -56,4 +68,158 @@ class SignUpEmailOTPTest : NativeAuthPublicClientApplicationAbstractTest() {
             }
         }
     }
+
+    /**
+     * Sign up with email + OTP. Resend email OTP.
+     * (hero scenario 1, use case 2.1.5, Test case 11)
+     */
+    @Test
+    fun testResendCode() {
+        config = getConfig(defaultConfigType)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        retryOperation {
+            runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+                val user = tempEmailApi.generateRandomEmailAddress()
+                val signUpResult = application.signUp(user)
+                assertResult<SignUpResult.CodeRequired>(signUpResult)
+                val otp1 = tempEmailApi.retrieveCodeFromInbox(user)
+                val codeRequiredState = (signUpResult as SignUpResult.CodeRequired).nextState
+                val resendCodeResult = codeRequiredState.resendCode()
+                assertResult<SignUpResendCodeResult.Success>(resendCodeResult)
+                val otp2 = tempEmailApi.retrieveCodeFromInbox(user)
+                Assert.assertNotEquals(otp1, otp2)
+            }
+        }
+    }
+
+    /**
+     * Sign up with email + OTP. User already exists with given email as email-otp account.
+     * (hero scenario 1, use case 2.1.6, Test case 12)
+     */
+    @Test
+    fun testErrorUserExistAsOTP() {
+        config = getConfig(ConfigType.SIGN_IN_OTP)
+        application = setupPCA(config, defaultChallengeTypes)
+
+        runBlocking { // Running with runBlocking to avoid default 10 second execution timeout.
+            val user = config.email
+            val signUpResult = application.signUp(user)
+            Assert.assertTrue(signUpResult is SignUpError)
+            Assert.assertTrue((signUpResult as SignUpError).isUserAlreadyExists())
+        }
+    }
+
+    /**
+     * Sign up with email + OTP. User already exists with given email as email-pw account.
+     * (Test case 13)
+     */
+    @Test
+    fun testErrorUserExistAsPassword() {
```

Comment:
This test doesn't belong here, and it already exists in `SignUpEmailPasswordTests` (test 1.1.10) so I think you can remove it.

Reply:
https://microsofteur-my.sharepoint.com/:w:/g/personal/sodenhoven_microsoft_com/ERrvxU_ZybBIvnq1UfVKWagBG-u_xFS7pcvRbVXCzFHnsQ?e=5EAUlc It was not included in the numbering of the Native Auth Acceptance criteria, so it was added to the Test plan later